python/packages/autogen-core/src/autogen_core/_agent_instantiation.py

```diff
@@ -118,3 +118,9 @@ def current_agent_id(cls) -> AgentId:
             raise RuntimeError(
                 "AgentInstantiationContext.agent_id() must be called within an instantiation context such as when the AgentRuntime is instantiating an agent. Mostly likely this was caused by directly instantiating an agent instead of using the AgentRuntime to do so."
             ) from e
+
+    @classmethod
+    def is_in_runtime(cls) -> bool:
```

Comment:
Consider renaming to be more clear
```suggestion
    def is_in_factory_call(cls) -> bool:
```